sklearn/tests/test_impute.py

```diff
@@ -635,3 +635,55 @@ def test_inconsistent_dtype_X_missing_values(imputer_constructor,
 
     with pytest.raises(ValueError, match=err_msg):
         imputer.fit_transform(X)
+
+
+@pytest.mark.parametrize("marker", [np.nan, -1, 0])
+def test_imputation_add_indicator(marker):
+    X = np.array([
+        [marker, 1, 5, marker, 1],
+        [2, marker, 1, marker, 2],
+        [6, 3, marker, marker, 3],
+        [1, 2, 9, marker, 4]
+    ])
+    X_true = np.array([
+        [3., 1., 5., 1., 1., 0., 0.],
+        [2., 2., 1., 2., 0., 1., 0.],
+        [6., 3., 5., 3., 0., 0., 1.],
+        [1., 2., 9., 4., 0., 0., 0.]
+    ])
+
+    imputer = SimpleImputer(missing_values=marker, add_indicator=True)
+    X_trans = imputer.fit_transform(X)
+
+    assert_allclose(X_trans, X_true)
+    assert_allclose(imputer.statistics_, np.array([3., 2., 5., np.nan, 2.5]))
```

Comment:
The invariant you want to hold is that the attributes of SimpleImputer(add_indicator=True) are identical to those of SimpleImputer(add_indicator=False), and their transform is identical except for some added columns. Ideally the test would be written in a way that expresses that invariant rather than repeating the expected values as it does here.